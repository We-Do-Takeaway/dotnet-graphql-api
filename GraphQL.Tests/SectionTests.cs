using System.Threading.Tasks;
using HotChocolate.Execution;
using Snapshooter.Xunit;
using Xunit;

namespace WeDoTakeawayAPI.GraphQL.Tests
{
    public class SectionTests : BaseTests
    {
        [Fact]
        public async Task Get_Section_By_Id()
        {
            var result = await ServiceProvider.ExecuteRequestAsync(
                QueryRequestBuilder
                    .New()
                    .SetQuery(@"
                        query SectionById($id: ID!) {
                          sectionById(id: $id) {
                            id
                            name
                          }
                        }")
                    .SetVariableValue(name: "id", value: "600dca30-c6e2-4035-ad15-783c122d6ea3")
                    .Create());
        
            result.MatchSnapshot();
        }

        [Fact]
        public async Task Get_Section_Items()
        {
            var result = await ServiceProvider.ExecuteRequestAsync(
                QueryRequestBuilder
                    .New()
                    .SetQuery(@"
                        query SectionById($id: ID!) {
                          sectionById(id: $id) {
                            id
                            name
                            items {
                              nodes {
                                id
                                name
                                description
                              } 
                            }
                          }
                        }")
                    .SetVariableValue(name: "id", value: "600dca30-c6e2-4035-ad15-783c122d6ea3")
                    .Create());
        
            result.MatchSnapshot();
        }
        
        [Fact]
        public async Task Get_Section_Menu()
        {
            var result = await ServiceProvider.ExecuteRequestAsync(
                QueryRequestBuilder
                    .New()
                    .SetQuery(@"
                        query SectionById($id: ID!) {
                          sectionById(id: $id) {
                            id
                            name
                            menu {
                              id
                              name
                              description
                            }
                          }
                        }")
                    .SetVariableValue(name: "id", value: "600dca30-c6e2-4035-ad15-783c122d6ea3")
                    .Create());
        
            result.MatchSnapshot();
        }
    }
}