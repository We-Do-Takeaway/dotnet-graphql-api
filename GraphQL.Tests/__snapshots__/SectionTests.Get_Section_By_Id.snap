{
  "Label": null,
  "Path": null,
  "Data": {
    "sectionById": {
      "id": "600dca30-c6e2-4035-ad15-783c122d6ea3",
      "name": "Desert"
    }
  },
  "Errors": null,
  "Extensions": null,
  "ContextData": null,
  "HasNext": null
}
